### PR TITLE
init

### PR DESCRIPTION
**Pull request by yosibarel**

### CODE REVIEW

File: app/src/androidTest/java/iband/com/ibandplayerdemo/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ * Instrumentation test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
```
Comment (moshest) [inline app/src/androidTest/java/iband/com/ibandplayerdemo/ExampleInstrumentedTest.java]: I think this file shouldn't be on the git repo.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            android:screenOrientation="landscape">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+
+        </activity>
+        <meta-data
+            android:name="stream_id"
+            android:value="5994219bbc454f6edf961315"/>
```
Comment (moshest) [inline app/src/main/AndroidManifest.xml]: Remove steam id and ast the user to put his own in "getting started" guide on the README file.

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+import com.iband.playersdk.IBandStream;
+
+import java.util.ArrayList;
+
+import iband.com.ibandplayerdemo.adapter.VariantAdapter;
+import iband.com.ibandplayerdemo.adapter.VariantHolder;
+
+public class PlayerActivity extends AppCompatActivity implements IBandPlayer.Listener, IBandStream.Listener {
+
+    private static final String STREAM_ID = "5994219bbc454f6edf961315";
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Duplicate, you already put stream ID on: app/src/main/AndroidManifest.xml

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+        if (state == IBandBasePlayer.State.READY) {
+            mProgressPlayer.setVisibility(View.GONE);
+            mViewPLayer.setVisibility(View.VISIBLE);
+            mBtnVariant.setVisibility(View.VISIBLE);
+            mViewPLayer.resetPanoramaPosition();
+        } else if (state == IBandBasePlayer.State.BUFFER || state == IBandBasePlayer.State.INITIALIZE)
+            mProgressPlayer.setVisibility(View.VISIBLE);
+        if (old == IBandBasePlayer.State.INITIALIZE) {
+            if (mStream.getStructure() == IBandStream.Structure.PLAIN) {
+                mViewPLayer.setState(IBandPlayerView.DisplayState.PLAIN);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Move those commands to an informative function such as `setupPlainView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+            setDisplayMode();
+            return;
+        }
+        super.onBackPressed();
+    }
+
+    @Override
+    public void onPlayerStateChanged(IBandPlayer iBandPlayer, IBandBasePlayer.State state, IBandBasePlayer.State old) {
+        if (state == IBandBasePlayer.State.READY) {
+            mProgressPlayer.setVisibility(View.GONE);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Move those commands to an informative function such as `initPlayerView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+            mViewPLayer.setVisibility(View.VISIBLE);
+            mBtnVariant.setVisibility(View.VISIBLE);
+            mViewPLayer.resetPanoramaPosition();
+        } else if (state == IBandBasePlayer.State.BUFFER || state == IBandBasePlayer.State.INITIALIZE)
+            mProgressPlayer.setVisibility(View.VISIBLE);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Move this command to an informative function such as `diplayBuffering()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+        if (old == IBandBasePlayer.State.INITIALIZE) {
+            if (mStream.getStructure() == IBandStream.Structure.PLAIN) {
+                mViewPLayer.setState(IBandPlayerView.DisplayState.PLAIN);
+                mBtnCenter.setVisibility(View.GONE);
+                mBtnVrMode.setVisibility(View.GONE);
+                mBtnVideoScale.setVisibility(View.VISIBLE);
+                stopThreadUpdateRotation();
+            } else if (mStream.getStructure() == IBandStream.Structure.EQUIRECTANGULAR) {
+                mViewPLayer.setState(IBandPlayerView.DisplayState.EQUIRECTANGULAR);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Move those commands to an informative function such as `setupEquirectangularView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+        mProgressStream.setVisibility(View.GONE);
+        toast(iBandError.getErrorMessage());
+    }
+
+    @Override
+    public void onStreamStateChanged(IBandStream iBandStream, IBandStream.State state, IBandStream.State old) {
+        if (old == IBandStream.State.INITIALIZE) {
+            mProgressStream.setVisibility(View.GONE);
+        }
+        if (iBandStream == mStream) {
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: No endless indentation, switch if case and `return`.

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+            if (iBandStream.getState() == IBandStream.State.CLOSED) {
+                mLvSettings.setAdapter(null);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: Move those commands to an informative function such as `setupEmptyView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+            mVariantAdapter.setSelected(position);
+            mPlayer.setCurrentVariant(mVariants.get(position).getIndex());
+        }
+        mVariantAdapter.notifyDataSetChanged();
+    }
+
+    private void setDisplayMode() {
+        if (mViewPLayer.getState() == IBandPlayerView.DisplayState.EQUIRECTANGULAR && mStream.getState() == IBandStream.State.OPEN) {
+            if (mViewPLayer.getVRMode()) {
+                mBottomActionBar.setVisibility(View.GONE);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java:340]: Move those commands to an informative function such as `setupEquirectangularVRView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+                mBtnPlayPause.setVisibility(View.GONE);
+                mLvSettings.setVisibility(View.GONE);
+                mBtnVrMode.setVisibility(View.GONE);
+                mBtnCenter.setVisibility(View.GONE);
+            } else {
+                mBottomActionBar.setVisibility(View.VISIBLE);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java:346]: Move those commands to an informative function such as `setupEquirectangularView()`

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+        mPlayer.addListener(this);
+        mPlayer.setStream(mStream);
+        if (mStream.getType() == IBandStream.Type.LIVE)
+            mViewLive.setVisibility(View.VISIBLE);
+        else
+            mViewLive.setVisibility(View.GONE);
+    }
+
+    private void initViews() {
+        mViewPLayer = (IBandPlayerView) findViewById(R.id.view_player);
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: This function is too long.
Split it to partial short methods.

File: app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        TextView tvVariant;
+        View vCurrentVariant;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        VariantHolder variant = getItem(position);
+        ViewHolder viewHolder;
+        if (convertView == null) {
+            viewHolder = new ViewHolder();
```
Comment (moshest) [inline app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java:37]: Move those commands to an informative function such as `setupView()`

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="app_name">IBand Live</string>
+    <string name="iband_token">5Dpf3hEbvuLdXBQs8hK9vjEJEdai8GaGDqne5CkomFXzczBFgopBMts</string>
```
Comment (moshest) [inline app/src/main/res/values/strings.xml]: Remove token and ast the user to put in in the "getting started" section on the README.

**Review by moshest — CHANGES_REQUESTED** [review]
Please see notes, and add getting started part on the README file.

File: app/src/androidTest/java/iband/com/ibandplayerdemo/ExampleInstrumentedTest.java (hunk 1 above)
Comment (yosibarel) [reply app/src/androidTest/java/iband/com/ibandplayerdemo/ExampleInstrumentedTest.java]: done

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (yosibarel) [reply app/src/main/AndroidManifest.xml]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 3 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 5 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 6 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 4 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 7 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 8 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 9 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 10 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java:340]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 11 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java:346]: done

File: app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java (hunk 12 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java]: done

File: app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java (hunk 13 above)
Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java:37]: done

Comment (yosibarel) [reply app/src/main/java/iband/com/ibandplayerdemo/adapter/VariantAdapter.java:37]: in this scope i will need to init the convertView and the view holder ... so i cannot to separate this block

File: README.md [hunk 15]
```diff
@@ -1,2 +1,13 @@
 # PlayerSDK-Android-Demo
 Demo player with IBandSDK player for Andorid.
+
+## TODO
```
Comment (moshest) [inline README.md]: Rename title to "How to Use"

File: README.md [hunk 16]
```diff
@@ -1,2 +1,13 @@
...
+
+**Token** Go to strings.xml and put the demo token into "iband_token" key.
+
+**StreamId** Go to PlayerActivity class and put the demo StreamId into STREAM_ID (located in the top of class).
+## Demo params
```
Comment (moshest) [inline README.md]: Delete this section (we will send token and streams to the client separately)

File: README.md [hunk 17]
```diff
@@ -1,2 +1,13 @@
...
+
+**Token** - 5Dpf3hEbvuLdXBQs8hK9vjEJEdai8GaGDqne5CkomFXzczBFgopBMts
+
+**StreamId** - 5994219bbc454f6edf961315
```
Comment (moshest) [inline README.md]: Add "Getting Started" section (copy steps from SDK readme).

**Review by moshest — CHANGES_REQUESTED** [review]

**Review by moshest — APPROVED** [review]

**Pull request merged**